lib/hci.c

```diff
@@ -303,20 +303,20 @@ static hci_map link_mode_map[] = {
 
 char *hci_lmtostr(unsigned int lm)
 {
-	char *s, *str = bt_malloc(130);
+	char *s, *str;
+	s = hci_bit2str(link_mode_map, lm);
+	if (!s) {
+		bt_free(str);
+		return NULL;
+	}
+	str = malloc(6 + strlen(s));
 	if (!str)
 		return NULL;
 
 	*str = 0;
 	if (!(lm & HCI_LM_MASTER))
 		strcpy(str, "SLAVE ");
 
-	s = hci_bit2str(link_mode_map, lm);
-	if (!s) {
-		bt_free(str);
-		return NULL;
-	}
-
 	strcat(str, s);
```

Comment:
may be we need to use strncat and also add one more line which sets str last character to null termination